File: app/src/main/java/com/example/soroushprofile/prefrences/widgets/SoroushPreference.java
package com.example.soroushprofile.prefrences.widgets;

import android.content.Context;
import android.graphics.Color;
import android.util.AttributeSet;
import android.widget.TextView;

import androidx.annotation.ColorInt;
import androidx.preference.Preference;
import androidx.preference.PreferenceViewHolder;

public class SoroushPreference extends Preference {

    private TextView mTitleView;

    @ColorInt
    private int textColor = Color.BLACK;

    public SoroushPreference(Context context) {
        super(context);
    }

    public SoroushPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public SoroushPreference(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public SoroushPreference(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }


    @Override
    public void onBindViewHolder(PreferenceViewHolder holder) {
        super.onBindViewHolder(holder);

        mTitleView = (TextView) holder.findViewById(android.R.id.title);

        setTitleColor(this.textColor);
    }

    public void setTitleColor(@ColorInt int color) {
        this.textColor = color;
        if (mTitleView == null) return;
        mTitleView.setTextColor(this.textColor);
    }
}

File: app/src/main/java/com/example/soroushprofile/prefrences/widgets/SoroushPreference.kt
package com.example.soroushprofile.prefrences.widgets

import android.content.Context
import android.graphics.Color
import android.util.AttributeSet
import android.widget.TextView

import androidx.annotation.ColorInt
import androidx.preference.Preference
import androidx.preference.PreferenceViewHolder

class SoroushPreference : Preference {

    private var mTitleView: TextView? = null

    @ColorInt
    private var textColor = Color.BLACK

    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {}


    override fun onBindViewHolder(holder: PreferenceViewHolder) {
        super.onBindViewHolder(holder)

        mTitleView = holder.findViewById(android.R.id.title) as TextView

        setTitleColor(this.textColor)
    }

    fun setTitleColor(@ColorInt color: Int) {
        this.textColor = color
        if (mTitleView == null) return
        mTitleView!!.setTextColor(this.textColor)
    }
}
